### PR TITLE
[runner.py] Do not silently ignore incorrect command line skips

### PR DESCRIPTION
**Pull request by tlively**
If `skip:some.test_thing` is passed on the command line, do not silently swallow
any errors that occur when trying to set it as skipped. This might help debug an
issue where the Windows roller is not correctly skipping a test that it is
supposed to be skipping.

### CODE REVIEW

File: tests/runner.py [hunk 1]
```diff
@@ -1761,16 +1761,19 @@ def skip_requested_tests(args, modules):
...
         print('will skip "%s"' % test, file=sys.stderr)
         suite_name, test_name = test.split('.')
         for m in modules:
           try:
             suite = getattr(m, suite_name)
             setattr(suite, test_name, lambda s: s.skipTest("requested to be skipped"))
+            skipped = True
             break
           except AttributeError:
             pass
```
Comment (sbc100) [inline tests/runner.py:1775]: Seems like this might be better written as: 

```
suite = getattr(m, suite_name, None)
if suite:
   setattr(suite, test_name, lambda s: s.skipTest("requested to be skipped"))
￼   skipped = True
￼   break
```

Rather then a try/catch... but unrelated this change I guess.

**Review by sbc100 — APPROVED** [review]

File: tests/runner.py (hunk 1 above)
Comment (tlively) [reply tests/runner.py]: The windows and mac circleci bots didn't seem to be running, so I took the opportunity to both make this change and try to get them going.

**Comment by tlively** [comment]
@sbc100 do you know what's up with the hanging bots?
